packages/server/lib/plugins/child/wrap-non-migrated-config.js

```diff
@@ -0,0 +1,58 @@
+const debugLib = require('debug')
+
+const debug = debugLib(`cypress:lifecycle:child:WrapNonMigrated:${process.pid}`)
+const { breakingOptions, breakingRootOptions } = require('@packages/config')
+
+/**
+ * Throw the error with the proper codeFrame
+ * @param {string} errorKey
+ * @param {string} name
+ */
+function throwInvalidOptionError (errorKey, name) {
+  debug('throwing err %s', name)
+  const errInternal = new Error()
+
+  Error.captureStackTrace(errInternal, throwInvalidOptionError)
+  const err = require('@packages/errors').getError(errorKey, { name, setupNodeEvents: true }, errInternal)
+
+  throw err
+}
+
+// only works if config.myProperty = 'something'
+// this will not throw config = {...config, myProperty: 'something'}
+function setInvalidPropSetterWarning (opts, errorKey, optionName, optionNameForError = optionName) {
```

Comment:
Doing these modifications at runtime, and not inside if `packages/config` doesn't seem ideal - I'm assuming you had a similar thought and this was the best solution, as opposed to adding this functionality in `packages/config`?

If we need to touch/modify this in the future, it'll be a bit painful to track down where these are added (who'd think to look here, lol).

Modifying the options via a reference passed in as an arg seems like such a hack, did you consider how we might declarative do this in `packages/config`?